src/index.ts

```diff
@@ -479,12 +484,14 @@ export function ConfirmMultisigAddress({
       // we're expecting this malleable object `multisig` that
       // gets passed in but really these interactions should
       // just get a braid or something derived from it.
-      const braid = Braid.fromData(JSON.parse(multisig.braidDetails));
+      const braidDetails: BraidDetails = JSON.parse(multisig.braidDetails);
+      const walletConfig = braidDetailsToWalletConfig(braidDetails);
       return new LedgerConfirmMultisigAddress({
+        ...walletConfig,
         expected: multisig.address,
         // this is for the name of the wallet the address being confirmed is from
         name,
-        braid,
+        braidIndex: Number(braidDetails.index),
```

Comment:
Is this clamp for precautionary measure because we don't have control over `braidDetails`? or have to work with pre-existing string values?

Reply:
Yeah, both. There are a lot of non-typescript files that will use this, so this is just a nice fallback. Also there are places where braidDetails comes as a json and then needs to be converted at the point it's passed in as well. I had this break on me without the `Number()` when testing with caravan so this just seemed safer. A lot a whack-a-mole when youre only partially typescript across a family of different libraries/projects.